bin/setup_stack

```diff
@@ -22,6 +22,13 @@ else
     shift
 fi
 
+if [ -z "$1" ]; then
```

Comment:
This should be something more specific, not just "is a variable defined" -- e.g. like "local" or "devices" -- also, put before $MODE.

Reply:
So this should be an non-optional argument?